packages/pluggableWidgets/notifications-native/src/Notifications.tsx

```diff
@@ -49,6 +57,17 @@ export function Notifications(props: NotificationsProps<undefined>): null {
             if (actions.length === 0) {
                 return;
             }
+
+            if (id !== undefined) {
+                if (knownIds.has(id)) {
+                    console.debug(`Not executing handleNotification for ${id}, has been executed before`);
+                    return;
+                } else {
+                    console.debug(`Executing handleNotification for ${id}`);
+                    knownIds.add(id);
```

Comment:
It's not recommended to mutate state like this, React will not integrate properly with it. Either use the second value from the state array `const [knownIds, setKnownIds]` and use that to update the state. An alternative would be to `useRef`. 

Which one to use depends:
- Should this `handleNotification` callback be re-computed when `knownIds` changes? 
- Should the component re-render when `knownIds` changes? 

If either should be the case, use `useState`. If neither should be the case, you can use `useRef`.

Reply:
done, replaced useState with useRef. Thank you